File: src/test/java/hw07/hw07Test.java
package hw07;

/**
 * Created by Antropov Igor on 31.10.2015.
 */

import org.junit.*;

public class hw07Test extends Assert {
    private OrderedArray<Character> testArray = new OrderedArray<Character>(1, 'z');
    private OrderedList<Character> testList = new OrderedList<Character>('z');
    private OrderedList<Character> wrongList = new OrderedList<Character>('X');
    private OrderedArray<Character> wrongArray = new OrderedArray<Character>(1, 'X');

    @Test
    public void orderedArrayTest() {
        testArray.add('a');
        testArray.add('c');
        testArray.add('w');
        testArray.add('f');
        testArray.add('s');
        testArray.add('d');
        assertEquals('a', (Object) testArray.get(0));
        assertEquals('d', (Object) testArray.get(2));
        assertEquals('z', (Object) testArray.get(6));
        assertEquals(7, testArray.size());
    }
    @Test
    public void orderedListTest() {
        testList.add('a');
        testList.add('c');
        testList.add('w');
        testList.add('f');
        testList.add('s');
        testList.add('d');
        assertEquals((Object) testList.get(0), 'a');
        assertEquals((Object) testList.get(2), 'd');
        assertEquals((Object) testList.get(6), 'z');
        assertEquals(testList.size(), 7);
    }
    @Test
    public void hashTest(){
        assertEquals(testArray.hashCode(), testList.hashCode());
        assertNotEquals(testArray.hashCode(), wrongArray.hashCode());
        assertNotEquals(testArray.hashCode(), wrongList.hashCode());
        assertNotEquals(testList.hashCode(), wrongArray.hashCode());
        assertNotEquals(testList.hashCode(), wrongList.hashCode());
    }
    @Test
    public void equalsTest(){
        boolean answer1 = testArray.equals(testList);
        boolean answer2 = testList.equals(testArray);
        boolean answer3 = testArray.equals(wrongArray);
        boolean answer4 = testArray.equals(wrongList);
        boolean answer5 = testList.equals(wrongArray);
        boolean answer6 = testList.equals(wrongList);
        assertEquals(true, answer1);
        assertEquals(true, answer2);
        assertEquals(false,answer3);
        assertEquals(false,answer4);
        assertEquals(false,answer5);
        assertEquals(false,answer6);
    }
    @Test
    public void compareToTest(){
        int answer1 = testArray.compareTo(testList);
        int answer2 = testList.compareTo(testArray);
        int answer3 = testArray.compareTo(wrongArray);
        int answer4 = testArray.compareTo(wrongList);
        int answer5 = testList.compareTo(wrongArray);
        int answer6 = testList.compareTo(wrongList);
        int answer7 = wrongArray.compareTo(testArray);
        int answer8 = wrongList.compareTo(testArray);
        assertEquals(0, answer1);
        assertEquals(0, answer2);
        assertEquals(1, answer3);
        assertEquals(1, answer4);
        assertEquals(1, answer5);
        assertEquals(1, answer6);
        assertEquals(-1, answer7);
        assertEquals(-1, answer8);
    }
}

File: src/test/kotlin/hw07/hw07Test.kt
package hw07

import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

/**
 * Created by Mikhail on 01.11.2015.
 */
public class HW07Test {
    @Test fun CompareToTest() {
        val arr1 = arrayOf(5, 4, 3, 11, 22);
        val arr2 = arrayOf(7, 6, 4, 33, 55, 44);
        val list1: OrderedList<Int> = OrderedList()
        val list2: OrderedList<Int> = OrderedList()
        for (i in 0..arr1.size - 1) {
            list1.add(arr1[i]);
            list2.add(arr1[i]);
        }
        assertTrue { list1.compareTo(list2) == 0}
        for (i in 0..arr2.size - 1) {
            list2.add(arr2[i]);
        }
        assertTrue { list1.compareTo(list2) < 0 }
    }

    @Test fun EqualsTest() {
        val arr1 = arrayOf('b', 'c', 'w' , 'q');
        val list1: OrderedList<Char> = OrderedList()
        val array1: ArrayOrderedList<Char> = ArrayOrderedList()
        val list2: OrderedList<Char> = OrderedList()
        for (i in 0..arr1.size - 1){
            list1.add(arr1[i])
            array1.add(arr1[i])
        }
        assertTrue  { list1.equals(array1) }
        assertFalse { list1.equals(list2) }
    }

    @Test fun HashCodeTest() {
        val arr1 = arrayOf(5, 4, 3);
        val list1: OrderedList<Int> = OrderedList()
        for (i in 0..arr1.size - 1) {
            list1.add(arr1[i]);
        }
        assertEquals(list1.hashCode(), 1684)
    }

    @Test fun OrderedListAddTest() {
        val arr1 = arrayOf(5, 44, 3);
        val list1: OrderedList<Int> = OrderedList()
        for (i in 0..arr1.size - 1) {
            list1.add(arr1[i]);
        }
        var str: String = ""
        for (i in 0..list1.size() - 1) {
            str += "${list1.get(i)}"
            str += " "
        }
        assertEquals("3 5 44 ", str)
    }
    @Test fun OrderedListContainsTest() {
        val arr1 = arrayOf(5, 44, 3);
        val list1: OrderedList<Int> = OrderedList()
        for (i in 0..arr1.size - 1) {
            list1.add(arr1[i]);
        }
        assertTrue { list1.contains(5) }
        assertFalse { list1.contains(66) }
    }
    @Test fun OrderedListSizeTest() {
        val arr1 = arrayOf(5, 44, 3);
        val list1: OrderedList<Int> = OrderedList()
        for (i in 0..arr1.size - 1) {
            list1.add(arr1[i]);
        }
        assertEquals(list1.size(), 3)
    }
    @Test fun ArrayOrderedListAddTest() {
        val arr1 = arrayOf(5, 44, 3);
        val array1: ArrayOrderedList<Int> = ArrayOrderedList()
        for (i in 0..arr1.size - 1) {
            array1.add(arr1[i]);
        }
        var str: String = ""
        for (i in 0..array1.size() - 1) {
            str += "${array1.get(i)}"
            str += " "
        }
        assertEquals("3 5 44 ", str)
    }
    @Test fun ArrayOrderedListContainsTest() {
        val arr1 = arrayOf(5, 44, 3);
        val array1: ArrayOrderedList<Int> = ArrayOrderedList()
        for (i in 0..arr1.size - 1) {
            array1.add(arr1[i]);
        }
        assertTrue { array1.contains(5) }
        assertFalse { array1.contains(66) }
    }
    @Test fun ArrayOrderedListSizeTest() {
        val arr1 = arrayOf(5, 44, 3);
        val array1: ArrayOrderedList<Int> = ArrayOrderedList()
        for (i in 0..arr1.size - 1) {
            array1.add(arr1[i]);
        }
        assertEquals(array1.size(), 3)
    }
}
